### PR TITLE
[packaging] Added conditional dependencies for wheel

### PR DESCRIPTION
**Pull request by ritwickdsouza**
References #72

### CODE REVIEW

**Comment by ritwickdsouza** [comment]
@nemesisdesign I ran the tests locally on a clean virtualenv and it passed.
On travis I'm getting this `ImportError: No module named 'jinja2'`

File: setup.py [hunk 1]
```diff
@@ -35,10 +41,6 @@ def get_install_requires():
             continue
         # add line to requirements
         requirements.append(line.replace('\n', ''))
-    # add py2-ipaddress if python2
```
Comment (nemesisdesign) [inline setup.py]: not sure this block of code can be removed, it may still be required for non-wheel packages.

Comment (nemesisdesign) [inline setup.py]: I'm testing this.

Comment (ritwickdsouza) [inline setup.py]: Ohh.

**Comment by nemesisdesign** [comment]
I can reproduce the travis issue wit the following procedure:

```bash
git clone https://github.com/ritwickdsouza/netjsonconfig.git tmp-netjsonconfig
cd tmp-netjsonconfig
git checkout cond-depend-wheel
mkvirtualenv tmp --python=/usr/bin/python3
pip install -U pip wheel setuptools
python setup.py develop
pip install -r requirements-test.txt
nosetests
```

File: setup.py (hunk 1 above)
Comment (nemesisdesign) [inline setup.py]: no, I'm wrong. Ignore this! Good work.

**Comment by nemesisdesign** [comment]
@ritwickdsouza keep me up to date

**Comment by ritwickdsouza** [comment]
@nemesisdesign I had introduced the bug by mistake. Works fine now.

File: setup.py (hunk 1 above)
Comment (nemesisdesign) [inline setup.py]: I was wrong again, that bit of code is necessary for non-wheel packages (old fashion tar.gz).

**Comment by nemesisdesign** [comment]
Also, please rebase from master with:

```shell
git checkout master
git pull https://github.com/openwisp/netjsonconfig.git master
git checkout cond-depend-wheel
git rebase master
```

File: setup.py (hunk 1 above)
Comment (ritwickdsouza) [inline setup.py]: How did you test it ?

Comment (nemesisdesign) [inline setup.py]: you can test it by generating the packages with:

```
python setup.py sdist bdist_wheel
```

Then create 2 new virtualenvs, one with python2 and one with python3, and in each virtualenv you do:

```
pip install -U pip wheel setuptools
pip install -r requirements-test.txt
```

Then, for each virtualenv you do:

```
cd dist/
pip install ./<name-of-tar.gz-package>
cd ..
./runtests.py
```

I did the same also for the wheel package (uninstalling the dependencies and reinstalling the package).

Without this test we would risk to release a broken package and would piss many people off!

Comment (ritwickdsouza) [inline setup.py]: Good to know. :+1:

**Pull request merged**